tasks/appsec.rake

```diff
@@ -6,10 +6,32 @@ namespace :appsec do
 
       version = args.to_a[0]
 
+      # You need to generate a token with the `repo` scope
+      # and configure SSO for DataDog's GitHub organization
+      token = ENV['GITHUB_TOKEN']
+
       ['recommended', 'strict'].each do |ruleset|
-        uri = URI("https://raw.githubusercontent.com/DataDog/appsec-event-rules/#{version}/build/#{ruleset}.json")
+        uri = URI("https://api.github.com/repos/DataDog/appsec-event-rules/contents/build/#{ruleset}.json?ref=#{version}")
+
+        http = Net::HTTP.new(uri.host, uri.port)
+        http.use_ssl = true
+
+        req = Net::HTTP::Get.new(uri)
+        req['Authorization'] = "Bearer #{token}"
+        req['Accept'] = 'application/vnd.github.raw+json'
 
-        File.open("lib/datadog/appsec/assets/waf_rules/#{ruleset}.json", 'wb') { |f| f << Net::HTTP::get(uri) }
+        http.request(req) do |res|
+          case res
+          when Net::HTTPSuccess
+            File.open("lib/datadog/appsec/assets/waf_rules/#{ruleset}.json", 'wb') do |f|
```

Comment:
Minor: Perhaps validate whether the file exist before sending http request. Kind of being protective when people refactor code and move files around.

Reply:
yes, it definitely makes sense to check

Reply:
Added here: [e3a7fcd](https://github.com/DataDog/dd-trace-rb/pull/3895/commits/e3a7fcd92fbb69dbed7e3fc83b2d63aefff9e7fd)